### PR TITLE
feat(Typography): Typography tweaks and format documentation

### PR DESCRIPTION
**Pull request by codecaaron**
## Changes
- Adds 3rd font size to Text.
- Adjusts line-height on all headings to a more appropriate 1.25.
- Adds custom margin intervals for headings.

## Additions
- Updates the storybook documentation to look nicer and more informative.

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.0.10-alpha.3caafc.0<br/>@codecademy/gamut@7.1.4-alpha.3caafc.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ea3153cadc7e879c3875536--codecademy-styleguide.netlify.app>https://5ea3153cadc7e879c3875536--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ea3153cadc7e879c3875536>Deploy Logs</a></p>

File: packages/gamut/src/Typography/styles/_variables.scss [hunk 1]
```diff
@@ -3,30 +3,37 @@
 $headingSizes: (
   "xxs": (
     "size": 16,
+    "margin": 16,
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Typography/styles/_variables.scss:6]: I thought we weren't going to add margins to text?

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/brand-components@1.2.1-alpha.371cce.0<br/>@codecademy/gamut@7.3.2-alpha.371cce.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://5ea9e5ccf87404fe8cf1e04d--codecademy-styleguide.netlify.app>https://5ea9e5ccf87404fe8cf1e04d--codecademy-styleguide.netlify.app</a></p><p><a href=https://app.netlify.com/sites/codecademy-styleguide/deploys/5ea9e5ccf87404fe8cf1e04d>Deploy Logs</a></p>

File: packages/gamut/src/Typography/styles/_variables.scss (hunk 1 above)
Comment (codecaaron) [reply packages/gamut/src/Typography/styles/_variables.scss:6]: It's currently optional.  I think potentially we want the regular behavior to be as text-flow like a possible then progressively remove those as we use them in other areas by disabling it (or the converse).

**Pull request merged**